### DISCUSSION TITLE
How to get UserLocation on top of other elements?

### DISCUSSION THREAD

**Discussion by Jobjeuh** [rnmapbox/maps#3087] [Q&A]
I have a couple of Points of Interest (POI) rendered as MarkerView:

```
<MapboxGL.MarkerView
   allowOverlap
   isSelected={currentPOI?.landMarker.name === POI.name}
   id={POI.name}
   key={POI.latitude + POI.longitude}
   coordinate={[POI.longitude, POI.latitude]}
>
     <Pressable onPress={() => setCurrentPOI({landMarker: POI, index})}>
          <View style={[POIMarker, markerStyle]} />
     </Pressable>
</MapboxGL.MarkerView>
```

Below that in the MapView I have a UserLocation:
```
<MapboxGL.UserLocation
    renderMode={UserLocationRenderMode.Native}
    showsUserHeadingIndicator
    onUpdate={handleLocationUpdate}
    minDisplacement={1}
    androidRenderMode="gps"
/>
```

This is what the order of the MapView looks like:
```
<MapboxGL.MapView
    localizeLabels
    rotateEnabled={false}
    logoEnabled={false}
    compassEnabled={false}
    zoomEnabled
    pitchEnabled
    scaleBarEnabled={false}
    style={styles.map}
    onTouchEnd={handleMapTouch}
>
    <Route />
    <POIs />
    <MapboxGL.Camera defaultSettings={defaultCameraSettings} ref={camera} />
    <MapboxGL.UserLocation
        renderMode={UserLocationRenderMode.Native}
        showsUserHeadingIndicator
        onUpdate={handleLocationUpdate}
        minDisplacement={1}
        androidRenderMode="gps"
    />
</MapboxGL.MapView>
```

See result:
![image](https://github.com/rnmapbox/maps/assets/68231796/67478355-0f4e-4e38-9e4c-5805528ef164)

How do I get the UserLocation on top of these MarkerViews?

Things I've tried:
- Changing the order in MapView
- Wrapping elements in ShapeSource and configuring `aboveLayerID` and `belowLayerID`
- Setting zIndex on elements

Is this something that is achievable? I've seen some issues mentioning this aswell, but they got closed with no further solutions

**Comment by mfazekas** [comment]
So MarkerViews are rendered on the top of the Map, not part of the map. So Marker Views are native views floating over the map.
While location component is part of the map. So you cannot place marker view behind elements that are part of the map.

Use SymbolLayer or PointAnnotatation

https://docs.mapbox.com/android/maps/api/10.16.1/mapbox-maps-android/com.mapbox.maps.plugin.locationcomponent/-location-component-constants/#com.mapbox.maps.plugin.locationcomponent/LocationComponentConstants/LOCATION_INDICATOR_LAYER/#/PointingToDeclaration/
